prow/apis/prowjobs/v1/types.go

```diff
@@ -350,7 +350,12 @@ func (rac *RerunAuthConfig) IsAllowAnyone() bool {
 }
 
 type ReporterConfig struct {
-	Slack *SlackReporterConfig `json:"slack,omitempty"`
+	ResultStore *ResultStoreReporter `json:"resultstore,omitempty"`
+	Slack       *SlackReporterConfig `json:"slack,omitempty"`
+}
+
+type ResultStoreReporter struct {
+	ProjectID string `json:"project_id,omitempty"`
```

Comment:
This value is never set because the type is only defined on the ProwJob resource, not in the job config.
Notice how the slack reporter configuration exists in central prow config for defaulting: https://github.com/kubernetes/test-infra/blob/139299a6007c3ef448adad5df389a8d30fdc4517/prow/config/config.go#L152
and also at the job level for job specific overrides: https://github.com/kubernetes/test-infra/blob/139299a6007c3ef448adad5df389a8d30fdc4517/prow/config/jobs.go#L123C5-L123C5

As mentioned before, I don't think we want to follow the same defaulting config pattern as the slack reporter. I think it would be more appropriate to add the fields to the `ProwJobDefault` struct which currently is only used for the `TenantID`: https://github.com/kubernetes/test-infra/blob/139299a6007c3ef448adad5df389a8d30fdc4517/prow/config/config.go#L219-L224C5

Reply:
Discussed offline. The type is actually defined in job config at the job level thanks to the ReporterConfig type being reused and copied as a whole from the config to the resource.
Higher level defaulting will be added in a later PR.